src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,691 @@
+    dsError_t status = dsERR_NONE;
+    dsFPDLedState_t iLedState;
+
+    status = dsFPGetSupportedLEDStates(&iLedState);
+    UT_LOG_INFO("Result dsFPGetSupportedLEDStates(OUT:uLEDState : %u)dsError_t:[%s]",iLedState,UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+
+    /* Check that the Indicator is valid */
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to set Power LED state.
+*
+* This test case provides a scope to  set the Power LED to differten FP States.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel indicator state should be set to ON through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 7 and provide the FP state to check.
+*
+*/
+void test_8_dsFPD_hal_SetLEDState(void)
+{
+    gTestID = 8;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    int32_t iLedState;
+
+    UT_LOG_MENU_INFO(" \t  Supported POWER LED States are:");
+    UT_LOG_MENU_INFO("------------------------------------------");
+
+    for (int fpState = dsFPD_LED_DEVICE_NONE; fpState < dsFPD_LED_DEVICE_MAX; fpState++)
+    {
+        UT_LOG_MENU_INFO("\t%d.  %-20s\n", fpState, UT_Control_GetMapString(dsFrontPanelLEDState, fpState));
+    }
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Select State: ");
+    scanf("%d", &iLedState);
+    readAndDiscardRestOfLine(stdin);
+    if(iLedState < dsFPD_LED_DEVICE_NONE || iLedState >=dsFPD_LED_DEVICE_MAX)
+    {
+        UT_LOG_INFO("Unsupported LED State");
+        goto exit;
+    }
+
+    /* Check that the Indicator is valid */
+    UT_LOG_INFO("Calling dsFPSetLEDState(IN state:[0x%X])",iLedState);
+    status = dsFPSetLEDState((dsFPDLedState_t)iLedState);
+    UT_LOG_INFO("Result dsFPSetLEDState()dsError_t:[%s]",UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+    exit:
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to get POWER LED State.
+*
+* This test case provides a scope to  get the State of FP Power LED.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel idicator state should be set to On through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 8 and provide indicator index.
+*
+*/
+void test_9_dsFPD_hal_GetLEDState(void)
+{
+    gTestID = 9;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    dsFPDLedState_t state = 0;
+
+    UT_LOG_INFO("Calling dsFPSetLEDState(OUT:state)");
```

Comment:
Here, function in log looks like wrong. Please replace with ' dsFPGetLEDState()'.

Reply:
updated